packages/tools/client-debugger/client-debugger-chrome-extension/src/react-components/AudienceView.tsx

```diff
@@ -25,21 +36,66 @@ export interface AudienceViewProps extends HasContainerId {
 export function AudienceView(props: AudienceViewProps): React.ReactElement {
 	const { containerId } = props;
 
-	// TODO: Post message requesting Audience summary
-	// TODO: Listen for Audience data updates
+	// POST Request for Audience Data
+	const getAudienceMessage: IDebuggerMessage = {
+		type: "GET_AUDIENCE_EVENT",
+		source: extensionMessageSource,
+		data: {
+			containerId,
+		},
+	};
+
+	const messageRelay = useMessageRelay();
+
+	const [clientId, setClientId] = React.useState<string[]>([]);
+	const [audienceState, setAudienceState] = React.useState<IClient[]>([]);
+	const [audienceHistory, setAudienceHistory] = React.useState<readonly AudienceChangeLogEntry[]>(
+		[],
+	);
 
 	React.useEffect(() => {
-		// TODO
-	}, [containerId]);
+		/**
+		 * Handlers for inbound messages related to Audience
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			["AUDIENCE_EVENT"]: (untypedMessage) => {
+				const message: AudienceEventMessage = untypedMessage as AudienceEventMessage;
+
+				setClientId(message.data.allAudienceClientId);
```

Comment:
I don't think this is right. `clientId` should be the local user's client ID, but isn't `allAudienceClientId` an array of all IDs?

Reply:
I suspect we need to broadcast the local clientId alongside the other audience data.

Reply:
Also note that since the client ID can change when the container connects / disconnects, we probably need to emit the audience change events from the debugger when the container disconnects and reconnects. You can compare the behavior between the extension and the embedded view in the sample app to make sure all of the updates line up between the two.

Reply:
As we are now message passing `audienceState` as `IClient[]` instead of `Map<string, IClient>`, I wanted to create a new array which contains the list of generated `clientId`. 

`useId` and `clientId` are different identifiers and as `clientId` isn't accessible through `IClient`, I wanted to propagate the list of `clientId` by retrieving the list of keys from `this.container.audience.getMembers()` ( which returns ` Map<string, IClient>`).


I might be missing something, but is there is a clever way that I can generate or access the locally generated `clientId` from `IClient` type?

Reply:
I believe the clientId (key from the map) is the same as `clientId.user.id`, unless I am mistaken.

Reply:
Now changed to `setClientId(message.data.clientId);`, which is from `clientId: this.container.clientId` in `FluidClientDebugger`.